nextflow/EVAImport/main.nf

```diff
@@ -0,0 +1,315 @@
+  val wait
+  path eva_script
+  val options
+  val merge_all_types
+  val fork
+  val sort_vf
+  val chr_synonyms
+  val remove_prefix
+  val output_file
+  
+  output: val 'ok'
+  
+  script:
+  def sort_vf_table     = sort_vf ? " --sort_vf" : ""
+  def merge_all         = merge_all_types ? " --merge_all_types" : ""
+  def use_fork          = fork ? "--fork ${fork}" : ""
+  def chr_synonyms_file = chr_synonyms ? " --chr_synonyms ${chr_synonyms}" : ""
+  def rm_prefix         = remove_prefix ? " --remove_prefix ${remove_prefix}" : ""
+  
+  """
+  perl ${eva_script} ${options} $sort_vf_table $merge_all $use_fork $chr_synonyms_file $rm_prefix --output_file ${output_file}
+  """
+}
+
+process run_variant_synonyms {
+  input:
+  val wait
+  path var_syn_script
+  val source_name
+  val species
+  val input_file
+  val registry
+  val host
+  val port
+  val dbname
+  
+  output: val 'ok'
+  
+  script:
+  
+  if(species == "sus_scrofa")
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+        perl ${var_syn_script} --source_name "pig_chip" --species ${species} --registry ${registry}
+      """
+  
+  else if(species == "rattus_norvegicus")
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+        perl ${var_syn_script} --source_name "rat" --species ${species} --registry ${registry} --host ${host} --port ${port} --user 'ensro' --db_name $dbname
+      """
+  else 
+      """
+        perl ${var_syn_script} --source_name ${source_name} --species ${species} --data_file ${input_file} --registry ${registry}
+      """
+}
+
+process run_variation_set {
+  input:
+  val wait
+  val my_species_set
+  val var_set_script
+  val files_path
+  val filenames
+  val species
+  val registry
+
+  output: val 'ok'
+
+  script:
+
+  def input_file = filenames.get(my_species_set)
+  """
+  perl ${var_set_script} -load_file ${files_path}${input_file} -registry ${registry} -species ${species} -variation_set ${my_species_set}
+  """
+}
+
+// Post-process variation feature sets
+// Populates variation_feature.variation_set_id
+process run_variation_set_2 {
+  input:
+  val wait
+  val var_set_script_2
+  val species
+  val registry
+
+  output: val 'ok'
+
+  script:
+
+  """
+  perl ${var_set_script_2} -registry_file ${registry} -species ${species}
+  """
+}
+
+process prepare_tables {
+  input:
+  val copy_script
+  val host
+  val dbname
+  val old_dbname
+
+  output: val 'ok'
+
+  script:
+  def new_host = "${host}-ensadmin"
+
+  """
+  sh ${copy_script} -h ${new_host} -o ${old_dbname} -n ${dbname}
+  """
+}
+
+process run_citations {
+  input:
+  val wait
+  path citations_script
+  val species
+  val registry
+  val file
+
+  output: val 'ok'
+
+  script:
+
+  """
+  perl ${citations_script} -load_file ${file} -registry ${registry} -species ${species}
+  """
+}
+
+
+workflow {
```

Comment:
Currently, the pipeline does not allow relative file paths for `params.registry` and `params.input_file`. You can support this by moving the `command_to_run` section of the code here:
```suggestion
workflow {
  // Build command to run EVA import
  registry = file(params.registry)
  input_file = file(params.input_file)

  command_to_run = " -i ${input_file} --source ${params.source} --source_description '${params.description}' --version ${params.version} --registry ${registry} --species ${params.species} --skip_tables '${params.skip_tables}'"

  log.info """
    Importing EVA data with the following parameters: \
    ${command_to_run}
  """
```

In this case, you would also need to use `registry` instead of `params.registry` as the arguments for functions in this workflow.